src/MiddlewareFactory.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import fetch from 'node-fetch';
+import { CustomFetchHandler, Middleware, RedirectHandler, RetryHandler, RetryHandlerOptions, TelemetryHandlerOptions } from "@microsoft/kiota-http-fetchlibrary";
+import { TelemetryHandler } from "@microsoft/kiota-http-fetchlibrary";
+import { ClientOptions } from "./IClientOptions";
+import { getGraphTelemetryConfigurator, GraphTelemetry } from "./middleware/TelemetryConfiguration";
+import { AuthenticationHandler } from "./middleware/AuthenticationHandler"
+import { GraphSDKConfig } from "./requestBuilderUtils/GraphSDKConfig";
+
+
+export function getDefaultMiddlewareChain(clientOptions: ClientOptions, graphSDKConfig?: GraphSDKConfig): Middleware[] {
+
+    const middlewareArray: Middleware[] = [];
+    if (!graphSDKConfig || !Object.keys(graphSDKConfig).length) {
+        const authenticationHandler = new AuthenticationHandler(clientOptions.authProvider);
```

Comment:
The getDefaultMiddlewareChain appends the authHandler only if the graphSDKConfig which is set by the service library is undefined or empty. 

This way we do not authenticate twice.

Reply:
thanks for the additional context. So does that mean depending on how the client gets initialized consumers will be able to use either the API method, or the fluent methods, but not both?

Reply:
We could keep a separate flag `appendAuthHandlerInMiddleware`. What do you think?

Reply:
I think we have better avenues here to avoid leaking that concern to the consumer.
With the request options : You could have a Graph request adapter, that inherits the kiota one, and override the send method simply to inject an option which says to the auth middleware "do nothing". This way the api method doesn't inject it, and the middleware does the auth. the send method inject it, and the middleware noops.
Without using options: we could update the middleware not to do any auth if the request header is already present.

Thoughts?

Reply:
You make a valid point that the implementation feels hacky here.
For now, I will verify if auth header is present and not empty and append only if that is the case.

Reply:
Changed this